proposals/lip-0052.md

```diff
@@ -362,712 +379,1832 @@ The module provides the following commands to modify the NFT store.
+        topics = []
+    )
+```
+
+#### removeSupportAllNFTs
+
+This function removes support for all non-native NFTs.
 
 ```python
-lock(moduleID, nftID):
-    nftID = canonicalNFTID(nftID)
-    if NFTStore(nftID).lockingModuleID != NFT_NOT_LOCKED:
-        lock fails
+def removeSupportAllNFTs() -> None:
 
-    NFTStore(nftID).lockingModuleID = moduleID
+    remove all entries from the supported NFTs substore
+
+    emitEvent(
+        module = MODULE_NAME_NFT,
+        name = EVENT_NAME_REMOVE_SUPPORT_ALL_NFTS,
+        data = EMPTY_BYTES,
+        topics = []
+    )
 ```
 
-#### unlock
+#### supportAllNFTsFromChain
 
-This function unlocks an NFT that was locked to a module ID.
+This function updates the supported NFTs substore to support all non-native NFTs of a specified foreign chain.
 
 ```python
-unlock(moduleID, nftID):
-    nftID = canonicalNFTID(nftID)
-    if NFTStore(nftID).lockingModuleID != moduleID:
-        unlock fails
+def supportAllNFTsFromChain(chainID: ChainID) -> None:
+
+    if there exists entry in the supported NFTs substore with key == ALL_SUPPORTED_NFTS_KEY:
+
+    if chainID == OWN_CHAIN_ID:
 
-    NFTStore(nftID).lockingModuleID = NFT_NOT_LOCKED
+    if supportedNFTsStore(chainID) exists:
+        supportedNFTsStore(chainID) = {"supportedNftIDArray": []}
+
+    else:
+        create substore entry with
+            substorePrefix = SUBSTORE_PREFIX_SUPPORTED_NFTS
+            key = chainID
+            value = encode(
+                schema = supportedNFTsStoreSchema,
+                object = {"supportedNftIDArray": []}
+            )
+
+    emitEvent(
+        module = MODULE_NAME_NFT,
+        name = EVENT_NAME_SUPPORT_ALL_NFTS_FROM_CHAIN,
+        data = {"chainID": chainID},
+        topics = [chainID]
+    )
 ```
 
-#### setAttributes
+#### removeSupportAllNFTsFromChain
 
-This function modifies the attributes of NFTs.
+This function removes support for all non-native NFTs of a specified foreign chain.
 
 ```python
-setAttributes(newAttributes, nftID):
-    nftID = canonicalNFTID(nftID)
-    if (NFTStore(nftID) does not exist
-        or length(newAttributes) > MAX_BYTE_LENGTH_ATTRIBUTES):
-        setAttributes fails
+def removeSupportAllNFTsFromChain(chainID: ChainID) -> None:
+
+    if there exists entry in the supported NFTs substore with key == ALL_SUPPORTED_NFTS_KEY:
+        raise Exception('Invalid operation. All NFTs from all chains are supported.')
 
-    NFTStore(nftID).attributes = newAttributes
+    if chainID == OWN_CHAIN_ID:
+        raise Exception('Invalid operation. Support for native NFTs cannot be removed.')
+
+    if supportedNFTsStore(chainID) does not exist:
+        return
+
+    delete entry supportedNFTsStore(chainID) from the supported NFTs substore
+
+    emitEvent(
+        module = MODULE_NAME_NFT,
+        name = EVENT_NAME_REMOVE_SUPPORT_ALL_NFTS_FROM_CHAIN,
+        data = {"chainID": chainID},
+        topics = [chainID]
+    )
 ```
 
-#### recover
+#### supportNftID
```

Comment:
To follow up on previous comment, I would think that supporting a certain collection rather than a specific nftID would be more useful